### PR TITLE
Corba / MQueue support 

### PR DESCRIPTION
**Pull request by ahoarau**
This completes https://github.com/orocos/rtt_geometry/pull/16.
- Jacobian and JntArray to the KDL Typekit
- CORBA support for KDL and Eigen types
- Mqueue support for KDL and Eigen types

### CODE REVIEW

**Comment by meyerj** [comment]
Looks good to me. https://github.com/orocos/rtt_ros_integration/pull/51 has been merged, too. I verified that rtt_geometry compiles successfully with this patch, but did not test it extensively yet.

What do you think, @smits?

File: eigen_typekit/corba/CorbaEigenConversion.hpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        //if(cols == 0) cols = 1; // For Vectors
+
+        cb.length( (CORBA::ULong)(tp.size() + 2) );
+        //log(Debug)<< "Total size for vector out  "<<cb.length() <<endlog();
+
+        cb[0] = static_cast<double>(rows);
+        cb[1] = static_cast<double>(cols);
+        //log(Debug)<< tp <<endlog();
+        Eigen::Map<Eigen::MatrixXd>(cb.get_buffer(/*CORBA::Boolean(false)*/) + 2 , rows, cols) = tp;
+        for(int i=0;i<cb.length();i++)
```
Comment (ahoarau) [inline eigen_typekit/corba/CorbaEigenConversion.hpp]: woops, L122 should be removed ! sorry about that..

**Comment by ahoarau** [comment]
There are a lot of comments to remove, but I thought it would be useful for you to check if everything's right.

**Comment by meyerj** [comment]
@smits?

**Comment by smits** [comment]
Please add newlines at EOF in the files that miss them. And line 122 of  eigen_typekit/corba/CorbaEigenConversion.hpp still needs to be removed.

**Comment by ahoarau** [comment]
ok I fixed the EOFs in the files and removed unnecessary comments everywhere.

**Comment by smits** [comment]
A small unittest that verifies correct nominal behaviour would be nice.

**Comment by meyerj** [comment]
Perhaps we can come up with a standardized typekit test suite defined directly in rtt or in a separate package that contains preprocessor macros, C++ template classes or fixtures and/or cmake macros to add tests for typekits and transports. This would avoid the need to redefine them in every package. Standardized tests could check all of the methods provided by the [TypeInfo](http://orocos-toolchain.github.io/devel/documentation/rtt/master/api/html/classRTT_1_1types_1_1TypeInfo.html) class interface and some out-of-band communication tests for all the transports found for the type. The only type-specific code that would need to be written is for loading the plugins and eventually for creating a non-default-initialized or random sample.

Just an idea. Otherwise it might be quite cumbersome to write a more or less complete test for every single typekit/transport plugin added or modified in the future. We do not have unit tests for typekits in rtt_ros_integration either.

**Comment by smits** [comment]
lgtm, @meyerj  we can live without a unit test for now?

**Comment by meyerj** [comment]
> lgtm, @meyerj we can live without a unit test for now?

I would say yes. The only code path worth to be tested is the conversion of C types to CORBA types or mqueue serialization and vice-versa. Everything else is more or less standard glue code or build instrumentation. Let's tackle this in a broader context for all typekit packages, as proposed above.

**Pull request merged**